src/components/Form/Form.jsx

```diff
@@ -1,21 +1,50 @@
 import React from 'react'
 import s from './Form.module.css'
+import { validateEmail, validateCPF } from './FormValidation.jsx'
+import { useState } from 'react'
 
 export const Form = () => {
+    const [input, setInput] = useState({})
+
+    const handleValidationResults = (event) => {
+        event.preventDefault();
+    
+        const { CPF, email } = input;
+        
+            if(!validateEmail(email) && !validateCPF(CPF)) {
+                alert('Revise os campos de e-mail e CPF. Suas informações são inválidas.');
+                return;
+              }
+              else if(!validateCPF(CPF)) {
+                alert('Insira um CPF válido.')
+              }
+              else if(!validateEmail(email)) {
+                alert('Insira um e-mail válido.')
+              }
+              else if(validateEmail(email) && validateCPF(CPF)) {
+                alert('Cadastro feito com sucesso');
+                window.location.reload();
+              }
+
+        
+      }
```

Comment:
Quando há um `return` na condição anterior, não precisamos usar `else` (pois se a execução chegou nessa linha, temos certeza que a condição anterior foi falsa). Um padrão que a gente encontra bastante é evitar if-else, justamente usando return.
Outro ponto é que se chegou no último if nesse caso, o email e o cpf serão válidos, pois você já checou nos ifs anteriores.
```suggestion
            if(!validateEmail(email) && !validateCPF(CPF)) {
                alert('Revise os campos de e-mail e CPF. Suas informações são inválidas.')
                return
              }
              if(!validateCPF(CPF)) {
                alert('Insira um CPF válido.')
                return
              }
              if(!validateEmail(email)) {
                alert('Insira um e-mail válido.')
                return
              }
            alert('Cadastro feito com sucesso')
            window.location.reload()
      }
```

Reply:
[Aqui](https://www.linkedin.com/pulse/avoid-if-else-hell-javascript-afsal-m-h/#:~:text=new%20Date().getDay())-,Early%20returns%20and%20less%20nesting,-Its%20a%20simple) uma referência sobre aproveitar o `return` para evitar aninhamento e `else`.

Reply:
No item 4 [desse post](https://www.linkedin.com/pulse/avoid-if-else-hell-javascript-afsal-m-h/) ele fala um pouco sobre essa técnica.